src/catch2/benchmark/catch_benchmark.hpp

```diff
@@ -96,7 +96,10 @@ namespace Catch {
             template <typename Fun, std::enable_if_t<!Detail::is_related<Fun, Benchmark>::value, int> = 0>
                 Benchmark & operator=(Fun func) {
                 fun = Detail::BenchmarkFunction(func);
-                run();
+                auto const* cfg = getCurrentContext().getConfig();
+                if (!cfg->skipBenchmarks()) {
+                    run();
+                }
```

Comment:
I think this should happen before we create the function wrapper.

Reply:
you'd place `fun = Detail::BenchmarkFunction(func);` inside the if-clause, did i got that right?

Reply:
Yeah. There is no reason to make the allocation if it won't be used afterwards.